scripts/conversion_script.py

```diff
@@ -482,6 +487,31 @@ def main():
             report=str(exception),
         )
     finally:
+        # Gather JSON & Textual report
+        data = gather_json_report()
+
+        if data:
+            highest_level = find_highest_report_level(actions=data["actions"])
+            # Set the first position of the list as being the final status,
+            # that's needed because `find_highest_report_level` will sort out
+            # the list with the highest priority first.
+            output.status = highest_level
+
+            if not output.message:
+                # Generate report message and transform the raw data into
+                # entries for Insights.
+                output.message, output.alert = generate_report_message(highest_level)
+
+            if not output.report:
+                # Try to attach the textual report in the report if we have
+                # json report, otherwise, we would overwrite the report raised
+                # by the exception.
+                output.report = gather_textual_report()
+
+            output.entries = transform_raw_data(data)
+
+        update_insights_inventory()
```

Comment:
This being here means that we will try to update insights even when the fail doesn't generate new report - maybe the preconversion/conversion didn't even run, is that okay? It shouldn't break anything :shrug: just additional call that may not bring new information.